### PR TITLE
Fix support for Django 4.0 and update test matrix

### PR DESCRIPTION
**Pull request by Stormheg**
This PR supersedes #72, commits retained because the original author deserves the attribution. Thanks for setting this in motion @AetherUnbound!

This PR updates the [django-bakery](https://github.com/palewire/django-bakery) dependency to 0.13.1 which includes support for Django 4.0. This means there is nothing holding us back anymore from supporting Django 4.0 and higher.

I've fixed the remaining compatibility issues with Django 4 and updated the testing matrix to the latest supported Django / Python / Wagtail versions.

Notably, this PR drops support for:
- Django 2.2, 3.0 and 3.1
- Wagtail 2.14 and older
- Python 3.6 and 3.7 (3.7 is quite old and not supported by Django 4.1)

Support for the following Django / Python / Wagtail versions has been added:
- Wagtail 2.15 LTS; v4.0 and the upcoming v4.1 LTS release.
  - v3.0 is not explicitly supported because it is near end of support but should probably work.
- Django 4.0 and 4.1
- Python 3.10

All tests pass for me locally but I have not given this a test drive in a real project. We should verify everything still works as expected before making a new release.

### CODE REVIEW

**Comment by AetherUnbound** [comment]
Thanks so much @Stormheg! 😄 Is there any chance you could publish a release candidate for this? I'd love to test it out on [the site I help run](https://github.com/pycascades/pycascades-cms) to see if it works before a full version release 🙂

**Comment by Stormheg** [comment]
Hi @AetherUnbound, I'd like to get a sanity-check / review from another core team member in first before publishing a release candidate. 

If you want to, you could start testing this right now from my branch. Just ```pip install git+https://github.com/Stormheg/wagtail-bakery.git@chore/django-4x-compat```.

Keen to hear about any issues you might encounter!

File: tox.ini
Comment (zerolab) [inline tox.ini:21]:
```suggestion
  wagtail41: wagtail>=4.1rc1,<4.2  # Upcoming LTS
```

should do the trick

File: setup.py
Comment (zerolab) [inline setup.py:44]: given this, should we mandata python>=3.8 in install requires?

**Review by zerolab — COMMENTED** [review]
Had a cursory look, no immediate issues spring to mind.
Made a couple of minor suggestions

**Comment by AetherUnbound** [comment]
Tested this on our project, the build command we use for testing didn't raise any errors! https://github.com/pycascades/pycascades-cms/actions/runs/3302171649/jobs/5448653445

File: tox.ini
Comment (Stormheg) [reply tox.ini:21]: That's pretty cool, did not know this was possible. Thanks for the suggestion Dan ❤️

I will get rid of the TODO comment above as well.

File: setup.py
Comment (Stormheg) [reply setup.py:44]: We don't currently mandate any Python version for wagtail-bakery. Python 3.7 should still work as long as you use a compatible Django.

I'm not entirely convinced mandating a Python version is strictly necessary, one should be able to use Python 3.7 (or even 3.6) with, say, Django 3.2 and Wagtail 2.15. Adding `python_requires='>=3.8'` would prevent one from doing this.

To me, the meaning of removing these versions from the classifiers means 'we don't guarantee support for those versions anymore, but feel free to try.'

My position is debatable of course. Happy to hear other insights on this subject.

Comment (zerolab) [reply setup.py:44]: Works for me

**Review by zerolab — APPROVED** [review]

**Comment by Stormheg** [comment]
Thanks for the review @zerolab 🙏

@AetherUnbound great to hear you are not encountering any issues!

There are some things in #74 I'd like to see in to the next release. Mostly some readme changes and a few setup.py tweaks that are visible on PyPI. Without committing to a timeline, you can expect a release 'soon'.

**Comment by Stormheg** [comment]
@AetherUnbound this is now released in [0.6.0]( https://github.com/wagtail/wagtail-bakery/releases/tag/0.6.0)!

**Comment by AetherUnbound** [comment]
Incredible, thank you!

**Pull request merged**
